### DISCUSSION TITLE
Testing Behavior of Generated Method with Dynamic Logic

### DISCUSSION THREAD

**Discussion by sharpzilla** [dotnet/roslyn#76496] [Q&A] [answered]
I am working on a source generator that uses an attribute to mark a type. The generator will then produce a method called:
```
Sum(int a, int b);
```

I need to write a unit test to verify that the Sum method works correctly and returns the expected result. The focus is on testing the behavior of the Sum function, not the functionality of the source generator itself, as I have several switches that can alter the behavior of the Sum function.

**Comment by CyrusNajmabadi** [comment, marked as answer]
Create a test library that references your generator.  Then create a unit test library that references your test library.  Have the unit test validate the 'sum' method that is generated into your library, just like you would write any other unit test of some piece of code.
